### PR TITLE
git auth token support for private repos

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/docker/actions-toolkit/pull/288

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,6 +117,22 @@ to the default Git context:
           push: true
 ```
 
+Building from the current repository automatically uses the [GitHub Token](https://docs.github.com/en/actions/security-guides/automatic-token-authentication),
+so it does not need to be passed. If you want to authenticate against another
+private repository for remote definitions, you can set the
+[`BUILDX_BAKE_GIT_AUTH_TOKEN` environment variable](https://docs.docker.com/build/building/variables/#buildx_bake_git_auth_token):
```
Comment (tonistiigi) [inline README.md:123]: Should there be a comment here that this requires buildx >= 0.14 ?

**Comment by crazy-max** [comment]
@dvdksn PTAL for copy thanks

File: README.md [hunk 2]
```diff
@@ -117,6 +117,25 @@ to the default Git context:
           push: true
 ```
 
+Building from the current repository automatically uses the [GitHub Token](https://docs.github.com/en/actions/security-guides/automatic-token-authentication),
+so it does not need to be passed. If you want to authenticate against another
```
Comment (dvdksn) [inline README.md:121]:
```suggestion
Building from the current repository automatically uses the `GITHUB_TOKEN` secret that GitHub [automatically creates for workflows](https://docs.github.com/en/actions/security-guides/automatic-token-authentication),
so you don't need to pass that manually. If you want to authenticate against another
```

File: README.md [hunk 3]
```diff
@@ -117,6 +117,25 @@ to the default Git context:
...
+        with:
+          source: "${{ github.server_url }}/${{ github.repository }}.git#${{ github.ref }}"
+          push: true
+        env:
+          BUILDX_BAKE_GIT_AUTH_TOKEN: ${{ secrets.MYTOKEN }}
+```
+
 ## Customizing
 
 ### inputs
```
Comment (dvdksn) [inline README.md:141]:
```suggestion
### Inputs
```

File: README.md [hunk 4]
```diff
@@ -138,20 +157,21 @@ The following inputs can be used as `step.with` keys
...
+| `provenance`   | Bool/String | [Provenance](https://docs.docker.com/build/attestations/slsa-provenance/) is a shorthand for `--set=*.attest=type=provenance`                                          |
+| `push`         | Bool        | Push is a shorthand for `--set=*.output=type=registry` (default `false`)                                                                                               |
+| `sbom`         | Bool/String | [SBOM](https://docs.docker.com/build/attestations/sbom/) is a shorthand for `--set=*.attest=type=sbom`                                                                 |
+| `set`          | List        | List of [targets values to override](https://docs.docker.com/engine/reference/commandline/buildx_bake/#set) (eg: `targetpattern.key=value`)                            |
+| `github-token` | String      | GitHub Token used to authenticate against a repository for [remote definitions](https://docs.docker.com/build/bake/remote-definition/) (default `${{ github.token }}`) |
```
Comment (dvdksn) [inline README.md:174]: shut this input be `git-tokens` or is it only for `GITHUB_TOKEN`?

File: README.md [hunk 5]
```diff
@@ -117,6 +117,25 @@ to the default Git context:
           push: true
 ```
 
+Building from the current repository automatically uses the [GitHub Token](https://docs.github.com/en/actions/security-guides/automatic-token-authentication),
+so it does not need to be passed. If you want to authenticate against another
+private repository for remote definitions, you can set the
+[`BUILDX_BAKE_GIT_AUTH_TOKEN` environment variable](https://docs.docker.com/build/building/variables/#buildx_bake_git_auth_token):
```
Comment (dvdksn) [inline README.md:123]:
```suggestion
[`BUILDX_BAKE_GIT_AUTH_TOKEN` environment variable](https://docs.docker.com/build/building/variables/#buildx_bake_git_auth_token)
or the `github-token` [input](#input).
```

File: README.md (hunk 3 above)
Comment (crazy-max) [reply README.md:141]: I use lowercasing here to match the `action.yml`: https://github.com/docker/bake-action/blob/75d6ba252c60fbd6a267235d93ee614a9b2d2826/action.yml#L9

I'm not against changing casing but would need to do that on all GHA repos we have. Can we look at this in follow-up?

File: README.md [hunk 6]
```diff
@@ -138,20 +158,21 @@ The following inputs can be used as `step.with` keys
...
+| `provenance`   | Bool/String | [Provenance](https://docs.docker.com/build/attestations/slsa-provenance/) is a shorthand for `--set=*.attest=type=provenance`                                          |
+| `push`         | Bool        | Push is a shorthand for `--set=*.output=type=registry` (default `false`)                                                                                               |
+| `sbom`         | Bool/String | [SBOM](https://docs.docker.com/build/attestations/sbom/) is a shorthand for `--set=*.attest=type=sbom`                                                                 |
+| `set`          | List        | List of [targets values to override](https://docs.docker.com/engine/reference/commandline/buildx_bake/#set) (eg: `targetpattern.key=value`)                            |
+| `github-token` | String      | GitHub Token used to authenticate against a repository for [remote definitions](https://docs.docker.com/build/bake/remote-definition/) (default `${{ github.token }}`) |
```
Comment (dvdksn) [inline README.md:175]:
```suggestion
| `github-token` | String      | API token used to authenticate to a Git repository for [remote definitions](https://docs.docker.com/build/bake/remote-definition/) (default `${{ github.token }}`) |
```

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**